lib/src/audio_source.dart

```diff
@@ -103,6 +122,186 @@ class AudioSource {
   Stream<void> get allInstancesFinished =>
       allInstancesFinishedController.stream;
 
+  // ///////////////////////////////////////
+  // / Filters for this [soundHash]
+  // ///////////////////////////////////////
+
+  /// Checks whether the given [filterType] is active.
+  ///
+  /// Returns `-1` if the filter is not active. Otherwise, returns
+  /// the index of the given filter.
+  int isFilterActive(FilterType filterType) {
+    final ret = SoLoudController().soLoudFFI.isFilterActive(
+          filterType,
+          soundHash: soundHash,
+        );
+    if (ret.error != PlayerErrors.noError) {
+      _log.severe(() => 'isFilterActive(): ${ret.error}');
+      throw SoLoudCppException.fromPlayerError(ret.error);
+    }
+    return ret.index;
+  }
+
+  /// Adds a [filterType] to this sound.
+  ///
+  /// Throws [SoLoudMaxFilterNumberReachedException] when the max number of
+  ///     concurrent filter is reached (default max filter is 8).
+  /// Throws [SoLoudFilterAlreadyAddedException] when trying to add a filter
+  ///     that has already been added.
+  PlayerErrors addFilter(FilterType filterType) {
+    final error = SoLoudController().soLoudFFI.addFilter(
+          filterType,
+          soundHash: soundHash,
+        );
+    if (error != PlayerErrors.noError) {
+      _log.severe(() => 'addGlobalFilter(): $error');
+      throw SoLoudCppException.fromPlayerError(error);
+    }
+    return error;
+  }
+
+  /// Removes [filterType] from all sounds.
+  PlayerErrors removeFilter(FilterType filterType) {
+    final error = SoLoudController().soLoudFFI.removeFilter(
+          filterType,
+          soundHash: soundHash,
+        );
+    if (error != PlayerErrors.noError) {
+      _log.severe(() => 'removeGlobalFilter(): $error');
+      throw SoLoudCppException.fromPlayerError(error);
+    }
+    return error;
+  }
+
+  /// Set the effect parameter with id [attributeId] of [filterType]
+  /// with [value] value.
+  ///
+  /// Specify the [attributeId] of the parameter (which you can learn from
+  /// [SoLoud.getFilterParamNames]), and its new [value].
+  ///
+  /// [handle] the handle to set the filter to. If equal to 0, the filter is
```

Comment:
This is weird. Either allow `null`, and then translate the null handle to `0` internally, or create two different methods (one for global filter and another one for single-voice filter).

Reply:
In `SoundHandle` fortunately we accept a value of 0, but this 0 is not handled by SoLoud c++ (only > 0). I used this only on the FFI side and this comment is wrong and you catched me! :) (I copied/pasted wrongly I think). It should be just:
`/// [handle] the handle to set the filter to.`

Here we are in the `AudioSource` and that `handle` is mandatory. While in `SoLoud.setGlobalFilterParameter()` is not a parameter but they both call the same FFI method.